app/controllers/pages_controller.rb

```diff
@@ -1,5 +1,62 @@
 class PagesController < ApplicationController
-  def activate
-    render template: "pages/#{params[:hpage]}"
+  #helper_method :update_permission
+
+  # setup all the new ActiveRecord objects that can be added
+  def admin
+    @game = @game.nil? ? Game.new : @game
```

Comment:
For game and proficiency, instead of doing a = a ? b : b, you can just do a ||= b.   #JustRubyThings